lib/new_playground.dart

```diff
@@ -562,44 +562,39 @@ class Playground implements GistContainer, GistController {
     }
   }
 
-  Future showNewFlutter() async {
+  Future<void> showNew(Layout layout) async {
     var loadResult = _loadGist();
     var shouldAutoRun = loadResult == LoadGistResult.storage;
 
     // If no gist was loaded, use a new Dart gist.
     if (loadResult == LoadGistResult.none) {
-      editableGist.setBackingGist(createSampleFlutterGist());
+      editableGist.setBackingGist(_createGist(layout));
     }
 
-    await _finalizeRoute(Layout.dart, shouldAutoRun);
-  }
-
-  Future showNewHtml() async {
-    var loadResult = _loadGist();
-    var shouldAutoRun = loadResult == LoadGistResult.storage;
-
-    // If no gist was loaded, use a new HTML gist.
-    if (loadResult == LoadGistResult.none) {
-      editableGist.setBackingGist(createSampleHtmlGist());
+    // Clear console output and update the layout if necessary.
+    var url = Uri.parse(window.location.toString());
+    _clearOutput();
+    _changeLayout(layout);
+    if (url.hasQuery && url.queryParameters['line'] != null) {
+      _jumpToLine(int.parse(url.queryParameters['line']));
     }
 
-    await _finalizeRoute(Layout.dart, shouldAutoRun);
+    await _analyzeAndRun(autoRun: shouldAutoRun);
   }
 
-  Future showNewDart() async {
-    var loadResult = _loadGist();
-    var shouldAutoRun = loadResult == LoadGistResult.storage;
-
-    // If no gist was loaded, use a new Dart gist.
-    if (loadResult == LoadGistResult.none) {
-      editableGist.setBackingGist(createSampleDartGist());
+  Gist _createGist(Layout layout) {
```

Comment:
nit: "createSampleGist" or "createDefaultGist" would be slightly more descriptive.